### ISSUE TITLE
Feature request/bug: Display 1 CD as a default when no "disc" tag is present

### ISSUE THREAD

**Issue by edhelas** [jcorporation/myMPD#1086]
In the Database view you can display the amount of discs the Albums actually have

![image](https://github.com/jcorporation/myMPD/assets/3726886/a89a7356-7436-4f95-a986-8f99377f4f07)

When there is only one CD it is quite common to not have any "disc" tag in the files. MyMPD actually display 0 CDs. 
It would be more convenient to actually display "1 CD" (also without the s).

Thanks for the awesome work !

**Comment by edhelas** [comment]
Hi, sorry to bother you again but I updated to the 11.0.3 version and it seems that I still have the counter set to 0 for some of my albums.

![image](https://github.com/jcorporation/myMPD/assets/3726886/6f9eab29-fc27-49f2-a859-9314a03e26c1)

Regards

**Comment by jcorporation** [comment]
You must force the recreation of the album cache. You can do it in the maintenance dialog.

**Comment by edhelas** [comment]
Thanks, will do :+1:

**Comment by mgonzalezm** [comment]
For me all the albums are shown as having '1 Disc', including the ones with more than 1 CD

All my music are FLAC files, and only the multi disc albums have the DISC=n vorbis tag. The disc tag is enabled in mpd.

what could be the problem?

**Comment by mgonzalezm** [comment]
Yes.

```
mpc --format='%disc% %track%_%title%' search '((artist == "Amy Winehouse") AND (album == "Back to Black"))'
1 1_Rehab
1 2_You Know I'm No Good
1 3_Me & Mr Jones
1 4_Just Friends
1 5_Back to Black
1 6_Love Is a Losing Game
1 7_Tears Dry on Their Own
1 8_Wake Up Alone
1 9_Some Unholy War
1 10_He Can Only Hold Her
1 11_Addicted
2 1_Valerie
2 2_Cupid
2 3_Monkey Man
2 4_Some Unholy War
2 5_Hey Little Rich Girl
2 6_To Know Him Is to Love Him
2 7_Love Is a Losing Game (original demo)
```

**Comment by jcorporation** [comment]
I will check this after my holiday.

**Comment by mgonzalezm** [comment]
Sure, no rush. Enjoy

In the mean time I did a little debugging, and found out that this function call is always returning NULL

https://github.com/jcorporation/myMPD/blob/304d1d2beb1f02b869c3abb85f38046041137ef8/src/lib/album_cache.c#L311

maybe the custom library libmpdclient that is included in the project in to blame?

**Comment by jcorporation** [comment]
Have you enabled the disc tag in myMPD?

**Comment by mgonzalezm** [comment]
Yes the disc tag is enabled in the settings.

In the album details it displays the disc number, but not when browsing.
![album](https://github.com/jcorporation/myMPD/assets/11181196/6e24a77d-899d-4ab0-ab6f-d36c98d1cba0)
![album-detail](https://github.com/jcorporation/myMPD/assets/11181196/b489fb71-a51b-4f02-99de-539f41c40cd9)

**Comment by jcorporation** [comment]
Ok, I think I know the root cause. myMPD does not enable the disc tag for building the album cache.

**Comment by GK-774** [comment]
"it is quite common to not have any "disc" tag in the files", yes correct, and if there are none then none should be displayed (e.g. with a line, like in the play view when a tag is enabled but not present).
Showing an invented disc number if there is none (and no physical CD involved) is not correct, IMHO.

**Comment by jcorporation** [comment]
I add an option how myMPD should interpret a missing disc tag.

**Comment by jcorporation** [comment]
Bug is fixed, the switch will be added for the 12.0.0 release.
